pkg/ddevapp/config.go

```diff
@@ -178,6 +177,29 @@ func (c *Config) Config() error {
 	return nil
 }
 
+// Validate ensures the configuraton meets ddev's requirements.
+func (c *Config) Validate() error {
+	// validate docroot
+	fullPath := filepath.Join(c.AppRoot, c.Docroot)
+	if _, err := os.Stat(fullPath); os.IsNotExist(err) {
+		return fmt.Errorf("no directory could be found at %s. Please enter a valid docroot in your configuration", fullPath)
```

Comment:
Absolutely love this. Thanks so much for adding it.

This might fall outside of this PR, but should we also use your app type detection and issue a warning if a known app type isn't detected/doesn't match configured app type?

Happy to push that one down the road if you would prefer, it's just a check that comes to mind as potentially useful as I look at these others.